adapters/oidc/js/src/main/resources/keycloak.js

```diff
@@ -1160,17 +1160,17 @@
                 return {
                     login: function(options) {
                         window.location.replace(kc.createLoginUrl(options));
-                        return createPromise().promise;
+                        return createPromise(true).promise;
```

Comment:
This is used both internally and externally. The login function should probably be extended with function(options, internal) where internal defaults to false. This will allow it to create the promise correctly for internal/external calls.